src/components/core/base/VisualizationLoadingHOC.tsx

```diff
@@ -79,22 +104,28 @@ export function visualizationLoadingHOC<T extends ICommonVisualizationProps & ID
     autoExecuteDataSource: boolean = true
 ): React.ComponentClass<T> {
     class LoadingHOCWrapped
-        extends React.Component<T & ILoadingInjectedProps, IVisualizationLoadingState> {
+        extends React.Component<T & ILoadingInjectedProps & IVisualizationLoadingProps, IVisualizationLoadingState> {
 
         public static defaultProps: Partial<T & ILoadingInjectedProps> = InnerComponent.defaultProps;
 
         protected subject: ISubject<IExecutionDataPromise>;
         protected pagePromises: IExecutionDataPromise[];
         protected hasUnmounted: boolean;
 
-        constructor(props: T & ILoadingInjectedProps) {
+        private sdk: SDK;
+
+        constructor(props: T & ILoadingInjectedProps & IVisualizationLoadingProps) {
             super(props);
 
             this.state = {
                 isLoading: false,
                 result: null,
                 error: null
             };
+
+            this.sdk =  props.sdk ? props.sdk.clone() : createSdk();
```

Comment:
other components consuming SDK object (eg. Visualization) handle also its change in componentWillReceiveProps

Reply:
Sorry. Could you explain more? I'm not sure I got your idea

Reply:
I am missing here something like here
https://github.com/gooddata/gooddata-react-components/blob/master/src/components/uri/Visualization.tsx#L241
to be consistent with other components which consume SDK as a prop

Reply:
thanks :)